### PR TITLE
Upgrade LavinMQ instances

### PR DESCRIPTION
**Pull request by viktorerlingsson**
### WHY are these changes introduced?

To upgrade LavinMQ instances. 

### WHAT is this pull request doing?

Adds upgrade_lavinmq resource and tests. 

### HOW can this pull request be tested?

Run VCR tests.

### CODE REVIEW

**Comment by tbroden84** [comment]
Great. We can also look into creating a fixtures for this. So the resource will get a VCR test and be tested on each commit.

**Comment by viktorerlingsson** [comment]
> Great. We can also look into creating a fixtures for this. So the resource will get a VCR test and be tested on each commit.

Something like this? https://github.com/cloudamqp/terraform-provider-cloudamqp/pull/296/commits/139e1106451f2faa835f6d9368115bd2a69c40ce

**Comment by tbroden84** [comment]
Looks good. Could also remove `TestAccUpgradeLavinMQ_Latest` test, should be the same as `TestAccUpgradeLavinMQ_Specific`. The latter we want to test.

**Comment by tbroden84** [comment]
We can also remove `test/configuration/insatnce_lavinmq_*.txt` files and remove the default values for subscription plan in `test/configuration/instance_*.txt`. This will force all other tests to specify the subscription plan to and we only need to have one configuration for `cloudamqp_instance` resouce..

**Comment by tbroden84** [comment]
Sweet. Also found a duplicated dependency that I removed.

File: docs/resources/upgrade_lavinmq.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+---
+
+# cloudamqp_upgrade_lavinmq
+
+This resource allows you to upgrade LavinMQ version. 
+
+There is two different ways to trigger the version upgrade
+
+> - Specify LavinMQ version to upgrade to
+> - Upgrade to latest LavinMQ version
```
Comment (tbroden84) [inline docs/resources/upgrade_lavinmq.md:15]: @viktorerlingsson Do we want to support upgrade to latest for LavinMQ? Been testing around a bit and will require some more changes to support that. If not we can remove some part of the documentation describing this.

Comment (viktorerlingsson) [reply docs/resources/upgrade_lavinmq.md]: No, you're right. I removed all text about upgrading to latest and re-phrased some stuff

**Pull request merged**